ush/python/pygfs/task/stage_ic.py

```diff
@@ -1,58 +1,485 @@
 
         Returns
         -------
-        None
+        Dict[str, Any]
+            Dictionary with base configuration values
         """
-        super().__init__(config)
+        base_keys = [
+            'RUN', 'MODE', 'EXP_WARM_START', 'NMEM_ENS',
+            'assim_freq', 'current_cycle', 'previous_cycle',
+            'ROTDIR', 'ICSDIR', 'STAGE_IC_YAML_TMPL', 'DO_JEDIATMVAR',
+            'OCNRES', 'waveGRD', 'ntiles', 'DOIAU',
+            'DO_JEDIOCNVAR', 'DO_STARTMEM_FROM_JEDIICE',
+            'DO_WAVE', 'DO_OCN', 'DO_ICE', 'DO_NEST', 'DO_CA', 'DO_AERO_ANL',
+            'USE_ATM_ENS_PERTURB_FILES', 'USE_OCN_ENS_PERTURB_FILES', 'DO_GSISOILDA', 'DO_LAND_IAU'
+        ]
+        return {key: self.task_config[key] for key in base_keys if key in self.task_config}
 
-    @logit(logger)
-    def execute_stage(self, stage_dict: Dict[str, Any]) -> None:
-        """Perform local staging of initial condition files.
+    def _get_config_vars(self) -> Dict[str, Any]:
+        """Calculate derived configuration variables
+
+        Returns
+        -------
+        Dict[str, Any]
+            Dictionary with derived configuration values (rRUN, OCNRES, START_ICE_FROM_ANA)
+        """
+        config_vars = {}
+
+        # Determine rRUN
+        config_vars['rRUN'] = "gdas" if self.task_config.RUN in ['gfs', 'gcafs'] else self.task_config.RUN
+
+        # OCNRES formatting
+        if "OCNRES" in self.task_config:
+            config_vars['OCNRES'] = f"{int(self.task_config.OCNRES):03d}"
+
+        # START_ICE_FROM_ANA logic
+        if self.task_config.get("DO_ICE", False):
+            config_vars['START_ICE_FROM_ANA'] = False
+            if self.task_config.get("DO_JEDIOCNVAR", False) and self.task_config.RUN == "gdas":
+                config_vars['START_ICE_FROM_ANA'] = True
+            if self.task_config.get("DO_STARTMEM_FROM_JEDIICE", False) and self.task_config.RUN == "enkfgdas":
+                config_vars['START_ICE_FROM_ANA'] = True
+
+        return config_vars
+
+    def _get_cycle_vars(self) -> Dict[str, Any]:
+        """Calculate current and previous cycle variables
+
+        Returns
+        -------
+        Dict[str, Any]
+            Dictionary with current and previous cycle variables including half_window
+        """
+        cycle_vars = {}
+        half_window = self.task_config.assim_freq // 2
+        cycle_vars['half_window'] = half_window
+
+        # Current cycle variables
+        if self.task_config.current_cycle:
+            cycle_vars['current_cycle_HH'] = self.task_config.current_cycle.strftime("%H")
+            cycle_vars['current_cycle_YMD'] = self.task_config.current_cycle.strftime("%Y%m%d")
+
+            if self.task_config.DOIAU and self.task_config.MODE == "cycled":
+                cycle_vars['model_start_date_current_cycle'] = self.task_config.current_cycle + timedelta(hours=-half_window)
+            else:
+                if self.task_config.get('REPLAY_ICS', False):
```

Comment:
`REPLAY_ICS` option was removed from `develop` in a previous PR.  This should be removed.